project/NativeImage.scala

```diff
@@ -16,6 +16,25 @@ object NativeImage {
     */
   private val includeDebugInfo: Boolean = false
 
+  /**
+   * List of classes that should be initialized at build time by the native image.
+   * Note that we strive to initialize as much classes during the native image build
+   * time as possible, as this reduces the time needed to start the native image.
+   * One wildcard could theoretically be used instead of the list, but to make things
+   * more explicit, we use the list.
+   */
+  private val defaultBuildTimeInitClasses = Seq(
```

Comment:
@JaroslavTulach Do you want to move this to `native-image.properties` inside `engine-runner` project? I think it is a bit better here, since all the other information for native image is kept in this source.

Reply:
I don't know.

All I see is that this is a long list with very wide wildcards - `scala`, `java`, `sun`! `com`! `org`! - if it works for our purposes, then great, but it is _blank check_ that matches almost everything.

Reply:
> but it is blank check that matches almost everything.

Isn't that the purpose? We want to match as many classes as possible for `--initialize-at-buildtime`.

I "tested" the native image on:
```
./runner --run engine/runner/src/test/resources/Factorial.enso 6
```
and it works as expected.

Reply:
@JaroslavTulach Do you have any other ideas what to test? Importing anything from `Standard.Base` does not work, neither on the latest `develop`. The size of the generated binary seems to be the same. The only other thing that is worth checking is code size and heap size (e.g. by GraalVM dashboard). Last time I have checked that, it did not work out of the box. I am uncertain whether I should spend any more time checking something on the native image. Can we merge this?

Reply:
> We want to match as many classes as possible for `--initialize-at-buildtime`.

As many Enso interpreter classes, true. I don't care about the other ones and certainly I _don't want to force them_ to be initialized in build time. I just want to _allow them to be initialized_ when they are needed transitively.

> Do you have any other ideas what to test?

There has always been just a single test:
```
./runner --run engine/runner/src/test/resources/Factorial.enso <a number>
```
that's all we have until #4877 or similar allows us to use standard libraries.

Reply:
> Can we merge this?

Of course! How can support for polyglot inner classes (hidden behind `@TruffleBoundary`)  influence native binary? 

> what to test? 
> - The size of the generated binary seems to be the same. 
> - code size and heap size

It is an interesting insight into our native image version of Enso, but...
 
> - I should spend any more time checking something on the native image. 

...I don't think it is relevant for this PR. Just merge.

Reply:
> As many Enso interpreter classes, true. I don't care about the other ones and certainly I don't want to force them to be initialized in build time. I just want to allow them to be initialized when they are needed transitively.

Right. This is precisely how this list originated - I added `org.enso` first to the `--initialize-at-buildtime`, and then added all the classes that are initialized from any class from `org.enso`. I understand that this list seems almost like a wildcard, but is there any other, simpler, solution?